lib/spack/spack/test/conftest.py

```diff
@@ -1706,3 +1706,54 @@ def noncyclical_dir_structure(tmpdir):
         with open(j('file_3'), 'wb'):
             pass
     yield d
+
+
+@pytest.fixture(scope='function')
+def mock_url_config(monkeypatch):
```

Comment:
I was under the impression you had a policy against mocking parts of methods (or was it modules) that are being tested.  If I mock that method then I am mocking part of the method being tested by `config.test_config_fetch_remote_configs`.

Regarding placement, this fixture is currently used in `test.cmd.env.test_env_include_packages_url` *and* `test.config.test_config_fetch_remote_configs`.

(FWIW. Although I haven't used it, I was wondering when I implemented this if it would be simpler if we included `pytest-mock`.)

Reply:
Added docstring.

Reply:
> I was under the impression you had a policy against mocking parts of methods (or was it modules) that are being tested.

My thought was to pull the `_fetch_file` definition into a top-level function, at which point it is not more-entangled than in other cases.

I missed that this is used in two testing modules so it does have to be here, but I still think it can be simplified: I think it is difficult to track that the tests interact with this fixture implicitly by defining URLs that contain the substring `"default"`, so if you have another preferred way to address that, then that works fine with me. I'm also happy to make suggestions on other possible ways to simplify this.

Reply:
Fair point.  I removed that (directory) check in favor of presence/absence of a file extension.